File: ui/src/main/java/com/peakmain/ui/widget/RoundCheckBox.java
package com.peakmain.ui.widget;

import android.content.Context;
import android.graphics.Canvas;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.AppCompatCheckBox;
import android.util.AttributeSet;

import com.peakmain.ui.R;

/**
 * author ：Peakmain
 * createTime：2020/3/27
 * mail:2726449200@qq.com
 * describe：
 */
public class RoundCheckBox extends AppCompatCheckBox {
    public RoundCheckBox(Context context) {
        this(context,null);
    }

    public RoundCheckBox(Context context, AttributeSet attrs) {
        this(context, attrs, android.support.v7.appcompat.R.attr.radioButtonStyle);
    }


    public RoundCheckBox(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        boolean checked = isChecked();
        if(!checked){
            setButtonTintList(ContextCompat.getColorStateList(getContext(), R.color.color_6A6A6A));
        }else{
            setButtonTintList(ContextCompat.getColorStateList(getContext(),R.color.color_08B906));
        }
    }
}

File: ui/src/main/java/com/peakmain/ui/widget/RoundCheckBox.kt
package com.peakmain.ui.widget

import android.content.Context
import android.graphics.Canvas
import android.os.Build
import android.util.AttributeSet
import androidx.annotation.RequiresApi
import androidx.appcompat.widget.AppCompatCheckBox
import androidx.core.content.ContextCompat
import com.peakmain.ui.R

/**
 * author ：Peakmain
 * createTime：2020/3/27
 * mail:2726449200@qq.com
 * describe：
 */
class RoundCheckBox @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = R.attr.radioButtonStyle) : AppCompatCheckBox(context, attrs, defStyleAttr) {
    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        val checked = isChecked
        buttonTintList = if (!checked) {
            ContextCompat.getColorStateList(context, com.peakmain.ui.R.color.ui_color_6A6A6A)
        } else {
            ContextCompat.getColorStateList(context, com.peakmain.ui.R.color.ui_color_08B906)
        }
    }
}